src/extensions/extension_manager/src/extension_manager_helper.cpp

```diff
@@ -120,19 +120,28 @@ ADUC_Result ProcessDownloadHandlerExtensibility(
 
 /**
  * @brief Get the download timeout in minutes from compile-time download options, or the override from config file.
- *
+ * @remark This function requires that the ADUC_ConfigInfo singleton has been initialized.
  * @param downloadOptions The download options.
  * @return unsigned int The download timeout that should be used.
  */
 unsigned int GetDownloadTimeoutInMinutes(const ExtensionManager_Download_Options* downloadOptions) noexcept
 {
-    ADUC_ConfigInfo config{};
-    if (!ADUC_ConfigInfo_Init(&config, ADUC_CONF_FILE_PATH) || (config.downloadTimeoutInMinutes == 0))
+    unsigned int ret = CONTENT_DOWNLOADER_MAX_TIMEOUT_IN_MINUTES_DEFAULT;
+    const ADUC_ConfigInfo* config = ADUC_ConfigInfo_GetInstance();
+    if (config == nullptr)
```

Comment:
probably better to crash if this is NULL, which it should never really be. systemd will restart anyways.

Reply:
I this case, if we're failing to get the download timeout value, we should fallback to the default value instead of crashing the agent. So that we don't see the agent keep crashing over and over.

Reply:
Yes, this makes sense in this case.